docs/requirements.md

```diff
@@ -1,16 +1,59 @@
-# PerfCheck CI Requirements
+# Perf Check CI Requirements
 
 Perf Check CI is a web application to manage jobs that measure and compare performance on another web application.
 
 ## Creating a job
 
-* As a user I want to create a job to compare the performance of two branches across multiple request paths.
+* As a user, I want to compare relative performance of my branch against master to make sure I'm not shipping code that will slow things down.
```

Comment:
Do you think we should configure a fixed reference branch or are you using `master` as shorthand for the branch that is deployed to production?

Reply:
Consistently using “reference branch” might make sense as you also use “reference measurement” to refer to the results of this branch further down in these requirements.

Reply:
Yes, I was also wondering what to do about about consistency issues. I voiced these drafts from the point of view of the user, who 95% of the time will only care about master and only in edge cases will care that the reference branch can be other branches. 

I’m fine with altering them to be consistent if that’s preferred over an “accurate” user point of view. *1 for “Reference Branch.”

In terms of defaults in the UI, we should absolutely be assuming they want to reference against master.

Reply:
Requirements are written from the perspective of the user or role by that doesn't mean it has to be in their voice or wording. I'm fine whatever we use as long as it makes the most sense or is usable and consistent and we all know what it means.

Wording in the requirements don't have to appear in the UI. However, if you're lucky you can have consistent naming across stories, implementation, and UI.

Reply:
>  doesn't mean it has to be in their voice or wording.

Yeah, this tripped me up a bit generally. A user isn't always going to know feature ABC or domain language XYZ. It seems strange to have requirements about the user written from the developer point of view and knowledge — we could easily misinterpret / misconstrue / fail to identify what the actual user point of view and task is — which I thought was part of the point of the exercise.

Reply:
>  It seems strange to have requirements about the user written from the developer point of view and knowledge

Why? User stories aren’t intended to be autobiographical. In most cases, the user’s perspective or motivation can be expressed more clearly and more precisely using language or knowledge this user might lack.

>  we could easily misinterpret / misconstrue / fail to identify what the actual user point of view and task is

In my experience, clear and consistent terminology in the requirements helps rather than hinder this.

Reply:
> Why?

My intuition is that we often work with a different set of assumptions and language than a user might, which puts as at risk of not properly addressing their assumptions and instead rolling with our own. By striving to keep language compatible and consistent with user expectations, we might better understand and solve for their requirement.

> clear and consistent terminology in the requirements helps

100% with you on this. There are always multiple ways to be consistent, each with their tradeoffs. Just trying to clarify which tradeoffs you prefer...

Reply:
As Manfred mentioned, it’s of course nice to have consistent naming across stories, implementation, and UI.

I think we’re all experienced enough to be able to translate in cases where we feel it makes sense to use more “formal” or more “developer-y” terminology in our requirements or in the source code, while we’re using more “informal” or longer, more explanatory UI copywriting for the user.

In this specific case, the other option is to consistently use the term “master” instead of “reference branch” in the requirements with a short note or definition below the user stories that “master” is used as a shorthand for “reference branch” and that this might be something else in some cases as Manfred mentions above.

As with a lot of things, it’s a matter of finding a balance that makes sense. :)

Reply:
Sounds good! 

I updated this example to keep the distinction in place, as it specifically for the "default reference branch" (most clearly expressed as "master") and updated other usages where it's not relevant to the more generic "reference branch."